src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.NetCore.cs

```diff
@@ -0,0 +1,47 @@
+#if NETCOREAPP
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Datadog.Trace.ClrProfiler.Managed.Loader
+{
+    /// <summary>
+    /// A class that attempts to load the Datadog.Trace.ClrProfiler.Managed .NET assembly.
+    /// </summary>
+    public partial class Startup
+    {
+        internal static System.Runtime.Loader.AssemblyLoadContext DependencyLoadContext { get; } = new ManagedProfilerAssemblyLoadContext();
+
+        private static string ResolveManagedProfilerDirectory()
+        {
+            string tracerFrameworkDirectory = "netstandard2.0";
+            var tracerHomeDirectory = Environment.GetEnvironmentVariable("DD_DOTNET_TRACER_HOME") ?? string.Empty;
+            return Path.Combine(tracerHomeDirectory, tracerFrameworkDirectory);
+        }
+
+        private static Assembly AssemblyResolve_ManagedProfilerDependencies(object sender, ResolveEventArgs args)
+        {
+            var assemblyName = new AssemblyName(args.Name);
+            var path = Path.Combine(ManagedProfilerDirectory, $"{assemblyName.Name}.dll");
+
+            if (assemblyName.Name.Contains("Datadog.Trace")
```

Comment:
nit: `StartsWith()` can be slightly faster than `Contains()` (worst case, the latter has to look at the entire string).

In either case, we should probably specify `StringComparer.OrdinalIgnoreCase`/`StringComparison.OrdinalIgnoreCase`.

Reply:
Fixed in 7a2a2ce4d640c750eca578a28ef7c32cde1eadb4 .